packages/@aws-cdk/aws-servicecatalog/lib/portfolio.ts

```diff
@@ -0,0 +1,246 @@
+
+  /**
+   * Associate portfolio with an IAM User.
+   * @param user an IAM user
+   */
+  giveAccessToUser(user: iam.IUser): void;
+
+  /**
+   * Associate portfolio with an IAM Group.
+   * @param group an IAM Group
+   */
+  giveAccessToGroup(group: iam.IGroup): void;
+
+  /**
+   * Share portfolio with another account.
+   * @param accountId AWS account to share portfolio with
+   * @param options Options for the initiate share
+   */
+  share(accountId: string, options?: PortfolioShareOptions): void;
+}
+
+/**
+ * Represents a Service Catalog Portfolio.
+ */
+abstract class PortfolioBase extends Resource implements IPortfolio {
+
+  /**
+   * The ARN of the portfolio.
+   */
+  public abstract readonly portfolioArn: string;
+
+  /**
+   * The Id of the portfolio.
+   */
+  public abstract readonly portfolioId: string;
+
+  /**
+   * Enable the Portfolio for a specific Role
+   */
+  public giveAccessToRole(role: iam.IRole) {
+    this.associatePrincipal(role.roleArn, role.node.addr);
+  }
+
+  /**
+   * Enable the Portfolio for a specific User
+   */
+  public giveAccessToUser(user: iam.IUser) {
+    this.associatePrincipal(user.userArn, user.node.addr);
+  }
+
+  /**
+   * Enable the Portfolio for a specific Group
+   */
+  public giveAccessToGroup(group: iam.IGroup) {
+    this.associatePrincipal(group.groupArn, group.node.addr);
+  }
+
+  /**
+   * Share the portfolio with a designated account.
+   */
+  public share(accountId: string, options?: PortfolioShareOptions) {
+    const hashId = this.getKeyForPortfolio(accountId);
+    new CfnPortfolioShare(this, `PortfolioShare${hashId}`, {
+      portfolioId: this.portfolioId,
+      accountId: accountId,
+      shareTagOptions: options?.shareTagOptions,
+      acceptLanguage: options?.acceptLanguage,
+    });
+  }
+
+  /**
+   * Associate a principal with the portfolio.
+   */
+  private associatePrincipal(principalArn: string, identifier: string) {
+    const hashId = this.getKeyForPortfolio(identifier);
+    new CfnPortfolioPrincipalAssociation(this, `PortolioPrincipalAssociation${hashId}`, {
+      portfolioId: this.portfolioId,
+      principalArn: principalArn,
+      principalType: PrincipalType.IAM,
+    });
+  }
+
+  /**
+   * Create a unique id based off the L1 CfnPortfolio or the arn of an imported portfolio.
+   */
+  protected abstract getKeyForPortfolio(value: string): string;
+}
+
+/**
+ * Properties for a Portfolio.
+ */
+export interface PortfolioProps {
+
+  /**
+   * The name of the portfolio.
+   */
+  readonly name: string;
+
+  /**
+   * The provider name.
+   */
+  readonly provider: string;
+
+  /**
+   * The accept language.
+   * @default - No accept language provided
+   */
+  readonly acceptLanguage?: AcceptLanguage;
+
+  /**
+   * Description for portfolio.
+   * @default - No description provided
+   */
+  readonly description?: string;
+}
+
+/**
+ * A Service Catalog portfolio.
+ */
+export class Portfolio extends PortfolioBase {
+
```

Comment:
thanks, is this something that could be caught by an eslint rule?

Reply:
Turns out, yes! https://eslint.org/docs/rules/padded-blocks

@dponzo any chance you could see if you can add this rule to the standard collection of ESLint rules CDK uses? It will for sure need to be done in a separate PR, as there will probably be many, many current violations (which, fortunately, can be automatically corrected by ESLint).

Our current ESLint configuration is here: https://github.com/aws/aws-cdk/blob/cbd755298f6ab73e604b96cab7d1e7c5f0dd8af1/tools/cdk-build-tools/config/eslintrc.js#L10-L10

Let me know!

Reply:
sure! ill look into this after this next commit

Reply:
Thanks!